### PR TITLE
Upload 1200/report improvements

### PR DESCRIPTION
**Pull request by cfarmer-fearless**
This sends two new reports to the PS API: Upload started and upload completed.  The upload started report is sent in the post-create hook, and the upload completed report is sent the post-finish hook.

### CODE REVIEW

File: upload-server/internal/metadata/metadata.go [hunk 1]
```diff
@@ -321,3 +321,66 @@ func (v *HookEventHandler) PostReceive(event handler.HookEvent, resp hooks.HookR
...
+	}
+
+	logger.Info("REPORT upload-completed", "report", report)
+	return v.Reporter.Publish(ctx, report)
+}
+
+func (v *HookEventHandler) PostCreate(event handler.HookEvent, resp hooks.HookResponse) (hooks.HookResponse, error) {
+	uploadId := Uid() // Assuming each upload gets a unique ID generated
+	logger.Info("Generated UUID for new upload", "UUID", uploadId)
+	event.Upload.ID = uploadId // Set the new upload ID
```
Comment (cfarmer-fearless) [inline upload-server/internal/metadata/metadata.go:374]: We def don't want to be updating the upload ID here.

File: upload-server/internal/metadata/metadata.go [hunk 2]
```diff
@@ -321,3 +321,66 @@ func (v *HookEventHandler) PostReceive(event handler.HookEvent, resp hooks.HookR
...
+		Content:         content,
+	}
+
+	logger.Info("REPORT upload-started", "report", report)
+	return v.Reporter.Publish(ctx, report)
+}
+
+func (v *HookEventHandler) ReportUploadCompleted(ctx context.Context, manifest map[string]string, uploadId string) error {
+	logger.Info("Attempting to report upload completed", "uploadId", uploadId)
+	content := &UploadStatusContent{
```
Comment (cfarmer-fearless) [inline upload-server/internal/metadata/metadata.go:350]: We need a new content type for these reports.  Should just have the schema version, schema name, and a status field that takes "success" or "fail"

**Review by USZ7 — APPROVED** [review]
lgtm!

File: upload-server/internal/metadata/metadata.go [hunk 3]
```diff
@@ -102,19 +102,27 @@ func (r *Report) Identifier() string {
 	return r.UploadID
 }
 
-type MetaDataVerifyContent struct {
+type ReportContent struct {
```
Comment (whytheplatypus) [inline upload-server/internal/metadata/metadata.go:105]: can you use this as an excuse to move this somewhere better? probably like under a `reports` module or something?

**Review by dancing-goats — APPROVED** [review]

**Pull request merged**